Pinta.Core/Classes/DocumentLayers.cs

```diff
@@ -251,6 +251,10 @@ public ImageSurface GetClippedLayer (int index)
 		var surf = CairoExtensions.CreateImageSurface (Format.Argb32, document.ImageSize.Width, document.ImageSize.Height);
 
 		var g = new Context (surf);
+
+		// Use the EvenOdd fill rule to account for "holes" in the selection
+		g.FillRule = FillRule.EvenOdd;
```

Comment:
Thanks for looking into this!

The fix looks like the correct change, but even better would be to replace these three lines with `document.Selection.Clip (g);` which wraps up all of this logic

https://github.com/PintaProject/Pinta/blob/deb29b3e1f4190c870063d255cb6764ea73bbc24/Pinta.Core/Classes/DocumentSelection.cs#L82

Reply:
Thanks for the feedback! I changed it to use the `Clip` function.